plasmapy/analysis/magnetics.py

```diff
@@ -0,0 +1,43 @@
+"""_summary_
+   
+"""
+__all__ = []
+
+from warnings import warn
+import scipy.integrate as sp
+from typing import Tuple
+import numpy as np
+
+
+def compute_bfield(bdot_data: np.ndarray, times: np.ndarray, loop_area: float) -> Tuple[np.ndarray, np.ndarray]:
+    """
+    Returns array of the magnetic field and the corresponding time array
+
+    Parameters
+    ----------
+    bdot_data: `numpy.ndarray`
+        A data array containing voltage fluctuations from a bdot probe. The
+        array values are proportional to time changing magnetic field via Faradays Law
+
+    times: `numpy.ndarray`
+        The time series to the data collection.
+
+    loop_area: float
+        The area through which the changing flux is measured. 
+
+    Returns
+    -------
+    field_arr: `numpy.ndarray`
+        The array containing the magnetic field fluctuations. 
+
+    new_time: `numpy.ndarray`
+        The corresponding time series.   
+    """
+    if len(bdot_data) != len(times):
+        raise Exception("length of time and voltage arrays in not equal\n")
+
+    bdot_data[:] = [x / loop_area for x in bdot_data]
+    field_arr = sp.cumtrapz(bdot_data, times)  # Tesla
+    new_time = times[1:]
```

Comment:
I definitely agree with Steve here: https://github.com/PlasmaPy/PlasmaPy/issues/1648#issuecomment-1200451551 that if we do `initial = 0` in `cumtrapz`, we don't need to modify the `times` array and could skip the whole `new_time` thing.